### PR TITLE
feat(ui): add Arabic localization support

### PR DESCRIPTION
**Pull request by pr-Mais**
## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change.*

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/flutterfire_ui/lib/src/i10n/lang/ar.dart [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+        'لقد أرسلنا إليك بريدًا إلكترونيًا يحتوي على رابط لإعادة تعيين كلمة المرور الخاصة بك. من فضلك تفقد بريدك الالكتروني',
+    this.forgotPasswordHintText =
+        'أدخل بريدك الإلكتروني وسنرسل لك رابطًا لإعادة تعيين كلمة مرورك',
+    this.emailLinkSignInButtonLabel = 'تسجيل الدخول باستخدام الرابط السحري',
+    this.signInWithEmailLinkViewTitleText =
+        'تسجيل الدخول باستخدام الرابط السحري',
+    this.signInWithEmailLinkSentText =
+        'أرسلنا لك رسال على البريد الإلكتروني تحتوي رابطاً سحرياً. تفقد صندوق رسائلك واضغط على الرابط لتسجيل الدخول',
+    this.sendLinkButtonLabel = 'أرسل الرابط السحري',
+    this.arrayLabel = 'array',
```
Comment (lesnitsky) [inline packages/flutterfire_ui/lib/src/i10n/lang/ar.dart:244]: @pr-Mais why did you ignore all labels below starting from this?

File: packages/flutterfire_ui/lib/src/i10n/lang/ar.dart [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    this.linkEmailButtonText = 'التالي',
+    this.signInWithPhoneButtonText = 'تسجيل الدخول برقم الهاتف',
+    this.signInWithGoogleButtonText = 'تسجيل الدخول باستخدام Google',
+    this.signInWithAppleButtonText = 'تسجيل الدخول باستخدام Apple',
+    this.signInWithTwitterButtonText = 'تسجيل الدخول باستخدام Twitter',
+    this.signInWithFacebookButtonText = 'تسجيل الدخول باستخدام Facebook',
+    this.phoneVerificationViewTitleText = 'أدخل رقم هاتفك',
+    this.verifyPhoneNumberButtonText = 'التالي',
+    this.verifyCodeButtonText = 'تحقق',
+    this.verifyingPhoneNumberViewTitle = 'أدخل الرمز المرسل كرسالة نصية',
```
Comment (maxzod) [inline packages/flutterfire_ui/lib/src/i10n/lang/ar.dart:187]: " رمز التحقق"

File: packages/flutterfire_ui/lib/src/i10n/lang/ar.dart (hunk 1 above)
Comment (pr-Mais) [reply packages/flutterfire_ui/lib/src/i10n/lang/ar.dart:244]: I thought it's used internally in the code and doesn't show in the UI, will update it

File: packages/flutterfire_ui/lib/src/i10n/lang/ar.dart [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+    this.signInActionText = 'تسجيل الدخول',
+    this.registerActionText = 'تسجيل جديد',
+    this.signInButtonText = 'تسجيل الدخول',
+    this.registerButtonText = 'تسجيل جديد',
+    this.linkEmailButtonText = 'التالي',
+    this.signInWithPhoneButtonText = 'تسجيل الدخول برقم الهاتف',
+    this.signInWithGoogleButtonText = 'تسجيل الدخول باستخدام Google',
+    this.signInWithAppleButtonText = 'تسجيل الدخول باستخدام Apple',
+    this.signInWithTwitterButtonText = 'تسجيل الدخول باستخدام Twitter',
+    this.signInWithFacebookButtonText = 'تسجيل الدخول باستخدام Facebook',
```
Comment (maxzod) [inline packages/flutterfire_ui/lib/src/i10n/lang/ar.dart:183]: اسماء الشركات هتكون بالانجليزي ؟

File: packages/flutterfire_ui/lib/src/i10n/lang/ar.dart [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    this.phoneVerificationViewTitleText = 'أدخل رقم هاتفك',
+    this.verifyPhoneNumberButtonText = 'التالي',
+    this.verifyCodeButtonText = 'تحقق',
+    this.verifyingPhoneNumberViewTitle = 'أدخل الرمز المرسل كرسالة نصية',
+    this.unknownError = 'حدث خطأ غير متوقع',
```
Comment (maxzod) [inline packages/flutterfire_ui/lib/src/i10n/lang/ar.dart:188]: حدث خطا ما

File: packages/flutterfire_ui/lib/src/i10n/lang/ar.dart [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+    this.smsAutoresolutionFailedError =
+        'حدث خطأ أثناء محاولة قراءة الرمز تلقائياً. رجاءً قم بإدخاله يدوياً',
+    this.smsCodeSentText = 'تم إرسال الرمز كرسالة نصية',
```
Comment (maxzod) [inline packages/flutterfire_ui/lib/src/i10n/lang/ar.dart:191]: رمز التحقق

File: packages/flutterfire_ui/lib/src/i10n/lang/ar.dart [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+    this.sendingSMSCodeText = 'جاري إرسال الرمز برسالة نصية...',
+    this.verifyingSMSCodeText = 'جاري التحقق من الرمز المرسل...',
+    this.enterSMSCodeText = 'أدخل الرمز المرسل',
+    this.emailIsRequiredErrorText = 'البريد الإلكتروني مطلوب',
+    this.isNotAValidEmailErrorText = 'رجاء قم بإدخال بريد إلكتروني صالح',
+    this.userNotFoundErrorText = 'هذا الحساب غير موجود',
+    this.emailTakenErrorText = 'الحساب الذي تحاول تسجيله موجود مسبقاً',
```
Comment (maxzod) [inline packages/flutterfire_ui/lib/src/i10n/lang/ar.dart:198]: البريد الالكتروني مستخدم من قبل

File: packages/flutterfire_ui/lib/src/i10n/lang/ar.dart [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+    this.accessDisabledErrorText = 'تم تعطيل الوصول إلى هذا الحساب مؤقتًا',
+    this.wrongOrNoPasswordErrorText =
+        'كلمة المرور غير صالحة أو أن هذا المستخدم ليس لديه كلمة مرور',
+    this.signInText = 'تسجيل الدخول',
+    this.registerText = 'تسجيل جديد',
```
Comment (maxzod) [inline packages/flutterfire_ui/lib/src/i10n/lang/ar.dart:203]: إنشاء حساب

File: packages/flutterfire_ui/lib/src/i10n/lang/ar.dart [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+        'كلمات المرور المدخلة غير متطابقة',
+    this.confirmPasswordInputLabel = 'تأكيد كلمة المرور',
+    this.forgotPasswordButtonLabel = 'نسيت كلمة المرور؟',
+    this.forgotPasswordViewTitle = 'استرجاع كلمة المرور المنسية',
+    this.resetPasswordButtonLabel = 'إعادة تعيين كلمة المرور',
+    this.verifyItsYouText = 'تحقق من هويتك',
+    this.differentMethodsSignInTitleText =
+        'استخدم إحدى الطرق التالية لتسجيل الدخول',
+    this.findProviderForEmailTitleText = 'أدخل بريدك الإلكتروني للمتابعة',
+    this.continueText = 'متابعة',
```
Comment (maxzod) [inline packages/flutterfire_ui/lib/src/i10n/lang/ar.dart:225]: متابعه  =>  follow
استمرار => continue

File: packages/flutterfire_ui/lib/src/i10n/lang/ar.dart [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+    this.countryCode = 'رمز الدولة',
+    this.codeRequiredErrorText = 'رمز الدولة مطلوب',
+    this.invalidCountryCode = 'رمز الدولة هذا غير صالح',
+    this.chooseACountry = 'قم باختيار الدولة',
```
Comment (maxzod) [inline packages/flutterfire_ui/lib/src/i10n/lang/ar.dart:229]: اختر الدولة

File: packages/flutterfire_ui/lib/src/i10n/lang/ar.dart (hunk 1 above)
Comment (maxzod) [inline packages/flutterfire_ui/lib/src/i10n/lang/ar.dart:244]: why leave it in English ?

File: packages/flutterfire_ui/lib/src/i10n/lang/ar.dart [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+    this.enableMoreSignInMethods = 'تفعيل المزيد من طرق تسجيل الدخول',
+    this.signInMethods = 'طرق تسجيل الدخول',
+    this.provideEmail = 'أدخل بريدك الإلكتروني وكلمة المرور',
+    this.goBackButtonLabel = 'عودة للخلف',
```
Comment (maxzod) [inline packages/flutterfire_ui/lib/src/i10n/lang/ar.dart:233]: الرجوع

**Review by maxzod — COMMENTED** [review]
"السحري"  is the the right word for magical but i think if we skipped the magical keyword it would be better 
لقد ارسلنا رابط تسجيل الدخول 
تسجيل الدخول عن طريق الرابط

File: packages/flutterfire_ui/lib/src/i10n/lang/ar.dart (hunk 3 above)
Comment (pr-Mais) [reply packages/flutterfire_ui/lib/src/i10n/lang/ar.dart:183]: I'm not fully sure, but for Apple, they complained about this once in one of my apps as it's, *I think*, mentioned in their guidelines (I couldn't find it now)

**Comment by pr-Mais** [comment]
@maxzod I'm not sure how to translate `map` tbh, I know it literally  means خريطة, so I don't know if this is the common translation in CS Arabic books.
Personally I would prefer it to stay English.

**Comment by maxzod** [comment]
@pr-Mais 
we could use  "كائن"  since map in `JS` is just a `Object`

**Review by Salakar — APPROVED** [review]

**Review by maxzod — APPROVED** [review]

**Pull request merged**